File: Eprodavnica-Backend/src/main/java/Eprodavnica/EprodavnicaBackend/model/Recenzija.java
package Eprodavnica.EprodavnicaBackend.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Recenzija {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column
    private int ocena;

    @Column
    private String komentar;

    @Column
    private Date datumPravljenja;

    @ManyToOne(cascade = CascadeType.REFRESH, fetch = FetchType.EAGER)
    @JoinColumn
    private Korisnik musterija;

    @ManyToOne(cascade = CascadeType.REFRESH, fetch = FetchType.EAGER)
    @JoinColumn
    private Produkt produkt;

}

File: Eprodavnica-Frontend/other/src/app/MODEL/Recenzija.ts
import { ProduktMini } from "./ProduktMini"

export class Recenzija {
    id: number
    ocena: number
    komentar: string
    datumPravljenja: Date
    emailMustarija: string
    produkt: ProduktMini
    constructor(
        id: number,
        ocena: number,
        komentar: string,
        datumPravljenja: Date,
        emailMustarija: string,
        produkt: ProduktMini
    ) {
        this.id = id
        this.ocena = ocena
        this.komentar = komentar
        this.datumPravljenja = datumPravljenja
        this.emailMustarija = emailMustarija
        this.produkt = produkt
    }
}